### ISSUE TITLE
Wrong vertical speed and 2D speed

### ISSUE THREAD

**Issue by sbonaime** [juanmcasillas/gopro2gpx#17]
Hi,
on my mac, with this Hero8 black footage 
https://drive.google.com/file/d/1632G6hBeecq4KH8ukYSYOSU1RvS-v53G/view?usp=sharing
 and using gopro2gpx and https://djitelemetryoverlay.com/srt-viewer I can see that the vertical speed is going from -448 to 335 km/h and the 2D speed reach 1576 km/h
<img width="409" alt="Capture d’écran 2021-05-04 à 04-05-21 14 49 05" src="https://user-images.githubusercontent.com/750709/117006008-2ea6b700-ace8-11eb-9cc8-e7dc451f4ec7.png">

With https://goprotelemetryextractor.com/free/ and the same footage  vertical speed is correct from  0 to 8.34 km/h and the 2D speed reach 29 km/h
<img width="405" alt="Capture d’écran 2021-05-04 à 04-05-21 14 52 57" src="https://user-images.githubusercontent.com/750709/117006192-6dd50800-ace8-11eb-9ab8-49e205bdaf5f.png">

here is my command line. I used #15 version but it did not change the problem
````
gopro2gpx  GH011227_saut.MP4 test
GPSFIX change to 3 [lock 3D (ok)]
-- stats -----------------
- Ok:                202
- GPSFIX=0 (bad):      0 (skipped: 0)
- Empty (No data):     0
Total points:        202
--------------------------
GPSFIX change to 3 [lock 3D (ok)]
-- stats -----------------
- Ok:                202
- GPSFIX=0 (bad):      0 (skipped: 0)
- Empty (No data):     0
Total points:        202
--------------------------
```

**Comment by Simske** [comment]
I just checked the elevation and speed data:
![image](https://user-images.githubusercontent.com/2445660/117011986-c0192780-acee-11eb-8628-4d0585b08604.png)
and everything seems fine there.

I'm guessing the problems comes from the time data: the time for every trackpoint is resolved to a second, but there are 20 trackpoints per second. When calculating the speed from elevation and location data this can lead to high numbers (or division by 0 error).
I don't know if the Gopro data has a higher time resolution which is just not outputted by gopro2gpx, otherwise you would need to just interpolate the time data inside of a second.

**Comment by sbonaime** [comment]
files size from 
gopro2gpx 58k
 https://goprotelemetryextractor.com/free/ 81k
In the begining of gopro2gpx gpx file, I can see that my camera is a   <name>gopro7-track</name>. With goprotelemetryextractor it is a <src>HERO8 Black</src>. 
It may explain why...
@Simske can you send my the gpx file you get ? Did you try to plot it with  https://djitelemetryoverlay.com/srt-viewer ?

**Comment by Simske** [comment]
>  @Simske can you send my the gpx file you get ? Did you try to plot it with https://djitelemetryoverlay.com/srt-viewer ?

Yes, when I put it in I get the same results as you, so I assume our outputs are the same.

I just compared the files from goprotelemetryextractor.com and gopro2gpx, the data for a single trackpoint look like this:
goprotelemetryextractor.com:
```
<trkpt lat="46.0098989" lon="-1.3977705">
    <ele>0.514</ele>
    <time>2020-10-29T11:03:38.558Z</time>
    <fix>3d</fix>
    <hdop>104</hdop>
    <cmt>altitude system: MSLV; 2dSpeed: 7.301; 3dSpeed: 7.31</cmt>
</trkpt>
```
gopro2gpx
```
<trkpt lat="46.0098989" lon="-1.3977705">
    <ele>0.514</ele>
    <time>2020-10-29T11:03:38Z</time>
    <extensions>
    <gpxtpx:TrackPointExtension>
        <gpxtpx:hr>0</gpxtpx:hr>
        <gpxtpx:cad>0</gpxtpx:cad>
        <gpxtpx:speed>7.301</gpxtpx:speed>
        <gpxtpx:distance>0</gpxtpx:distance>
        </gpxtpx:TrackPointExtension>
    <gpxx:TrackPointExtension/>
    </extensions>
</trkpt>
```

As you can see from this:
- I was right about the timestamps, resolution of 1 second is not enough
- Speed and other additional data is written into a comment tag for goprotelemetryextractor, while gopro2gpx uses the Garmin GPX-extension to save only the speed.

I haven't worked directly on the decoding part of this package, maybe @juanmcasillas has an easy fix.

**Comment by sbonaime** [comment]
@Simske  I am diving into the code and I see a problem. The GPS decoded data we get does not have any millisecond data from Hero8 MP4 thus we have 20 points with the same datetime as you saw.

**Comment by younes-mirinezhad** [comment]
> @Simske I am diving into the code and I see a problem. The GPS decoded data we get does not have any millisecond data from Hero8 MP4 thus we have 20 points with the same datetime as you saw.

I have the same problem.
is there any solution?

**Comment by Simske** [comment]
@younes-mirinezhad No sorry, I didn't have time to look at it yet

**Comment by JimmyS83** [comment]
Could you try with PR#24 ?

**Comment by FreudBH** [comment]
Hello!
I noticed in the Update section "Added support for 18Hz resolution in GPS5 fourCC, to fix (#PR23,#PR17)[https://github.com/juanmcasillas/gopro2gpx/pull/24]. Thanks to (JimmyS83)[https://github.com/JimmyS83] for spot the problem."

Nevertheless I just run the last version and the speed is being calculated wrong. I am using GoPro Max 360.

[GH010015_1717275916188.gpx.txt](https://github.com/user-attachments/files/15749637/GH010015_1717275916188.gpx.txt)


![gopro2gpx speed wrong](https://github.com/juanmcasillas/gopro2gpx/assets/119543269/6329ed7d-4bfc-4702-8bdc-4071e82a5cd2)
